lib/3scale/backend/transactor/notify_batcher.rb

```diff
@@ -12,21 +12,17 @@ module NotifyBatcher
         include Resque::Helpers
         include Configurable
 
-        METRIC_AUTHORIZE = 'transactions/authorize'.freeze
-        METRIC_TRANSACTIONS = 'transactions'.freeze
-        private_constant :METRIC_AUTHORIZE, :METRIC_TRANSACTIONS
-
         def notify_authorize(provider_key)
-          notify(provider_key, METRIC_AUTHORIZE => 1)
+          notify(provider_key, configuration.master.metrics.transactions_authorize => 1)
```

Comment:
Have you considered keeping the constants?
```ruby
METRIC_AUTHORIZE = configuration.master.metrics.transactions_authorize
METRIC_TRANSACTIONS = configuration.master.metrics.transactions
private_constant :METRIC_AUTHORIZE, :METRIC_TRANSACTIONS
```

Reply:
I am not sure it is a good pattern to mix private constants and configuration parameters

Reply:
but those are not constants, those are method calls that could return anything at invocation time - setting those as constants makes two assumptions:
- at load time of this file the methods will return something sane and validated.
- the values will never change after they are obtained

Both are very risky assumptions.

Reply:
I agree with you that are risky assumptions and I thought the same regarding the second point you mentioned but after commeting it with davidor it's improbable for this values to change because if we changed this in the middle of the execution how would the other components respond?

If we cannot afford this assumptions then I agree we should not use the constants, because other factor that I thought that could be risky is the order on which the configuration is loaded.